### PR TITLE
Picsure 355 how to evaluate gates

### PR DESCRIPTION
**Pull request by YuAvillachLab**
Has passed its own unit test
Tested with copdgene, everything now acts as what it should be to fulfill all current requirements.

### CODE REVIEW

**Comment by gaborkorodi-hms** [comment]
I am holding off on reviewing this, until you let me know

**Comment by gaborkorodi-hms** [comment]
I am holding off on this until you let me know

File: pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/auth/AuthorizationService.java [hunk 1]
```diff
@@ -192,6 +199,11 @@ public boolean isAuthorized(Application application , Object requestBody, UUID u
                 }
             }
 
+            keys.add((accessRule.getCheckMapKeyOnly()==null)?"null":Boolean.toString(accessRule.getCheckMapKeyOnly()));
```
Comment (gkorodi) [inline pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/auth/AuthorizationService.java]: Don't like this. This is expensive operation and you are doing it twice! Don't like it! Would looooove to add a ticket to make this more efficient (IN THE FUTURE)

File: pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/auth/AuthorizationService.java [hunk 2]
```diff
@@ -191,6 +198,15 @@ public boolean isAuthorized(Application application , Object requestBody, UUID u
                     keys.add(subAccessRule.getRule());
                 }
             }
+            Boolean checkMapKeyOnly = accessRule.getCheckMapKeyOnly(),
```
Comment (gkorodi) [inline pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/auth/AuthorizationService.java]: Thank YOU Yu!

**Pull request merged**